Bottom: 1142d3a2be9531e93fb1cb5f7229efe76c67fd7b
Top:    67b3e606bdad2341c99aa72310b18dd933e60652
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-16 16:10:34 +0100

Refresh of work-in-progress-on-subcomposi

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index ae93619..2f30d4d 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -850,10 +850,15 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
                 1.);
 
   glDepthMask (0);
+  glStencilMask (~0);
   glClearStencil (0);
   glClear (GL_COLOR_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
   hidgl_reset_stencil_usage ();
 
+  /* Disable the stencil test until we need it - otherwise it gets dirty */
+  glDisable (GL_STENCIL_TEST);
+  glStencilFunc (GL_ALWAYS, 0, 0);
+
   region.X1 = MIN (Px (ev->area.x), Px (ev->area.x + ev->area.width + 1));
   region.X2 = MAX (Px (ev->area.x), Px (ev->area.x + ev->area.width + 1));
   region.Y1 = MIN (Py (ev->area.y), Py (ev->area.y + ev->area.height + 1));
